CRM/Sepa/Logic/Batching.php

```diff
@@ -515,124 +564,99 @@ protected static function syncGroups(
       }
 
       foreach ($financial_type_groups as $financial_type_id => $mandates) {
-        if (0 === $financial_type_id) {
-          $financial_type_id = NULL;
-        }
-        if (!isset($existing_groups[$collection_date][$financial_type_id ?? 0])) {
-          // this group does not yet exist -> create
-
-          // find unused reference
-          $reference = self::getTransactionGroupReference($creditor_id, $mode, $collection_date, $financial_type_id);
-
-          $group = civicrm_api('SepaTransactionGroup', 'create', array(
-            'version'                 => 3,
-            'reference'               => $reference,
-            'type'                    => $mode,
-            'collection_date'         => $collection_date,
-            'financial_type_id'       => $financial_type_id,
-            'latest_submission_date'  => date('Y-m-d', strtotime("-$notice days", strtotime($collection_date))),
-            'created_date'            => date('Y-m-d'),
-            'status_id'               => $group_status_id_open,
-            'sdd_creditor_id'         => $creditor_id,
-          ));
-          if (!empty($group['is_error'])) {
-            // TODO: Error handling
-            Civi::log()->debug("org.project60.sepa: batching:syncGroups/createGroup ".$group['error_message']);
+        $group_id = self::getOrCreateTransactionGroup(
+          (int) $creditor_id,
+          $mode,
+          $collection_date,
+          0 === $financial_type_id ? NULL : $financial_type_id,
+          (int) $notice,
+          $existing_groups
+        );
+
+        // now we have the right group. Prepare some parameters...
+        $entity_ids = [];
+        foreach ($mandates as $mandate) {
+          // remark: "mandate_entity_id" in this case means the contribution ID
+          if (empty($mandate['mandate_entity_id'])) {
+            // this shouldn't happen
+            Civi::log()
+              ->debug("org.project60.sepa: batching:syncGroups mandate with bad mandate_entity_id ignored:" . $mandate['mandate_id']);
```

Comment:
Shouldn't it be at least a warning, maybe even an error?

There should be a space after the `:`.